<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>FastaWriterHelper.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_1'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_2'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_3'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_4'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_5'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_6'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_7'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_8'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_9'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_10'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_11'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_12'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_13'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_14'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_15'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_16'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_17'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_18'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_19'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_20'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Created on 01-21-2010</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_21'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_22'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence.io;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_23'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_24'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.DNASequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_25'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.GeneSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_26'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.ProteinSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_27'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_28'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_29'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.io.template.FastaHeaderFormatInterface;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_30'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.Compound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_31'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.Sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_32'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_33'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.BufferedOutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_34'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.File;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_35'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.FileOutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_36'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.OutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_37'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.ArrayList;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_38'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.Collection;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_39'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_40'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_41'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * The class that should be used to write out fasta file of a sequence collection</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_42'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Scooter Willis &#60;willishf at gmail dot com&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_43'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_44'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>public class FastaWriterHelper {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_45'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_46'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_47'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_48'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_49'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write collection of protein sequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_50'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_51'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_52'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param proteinSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_53'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_54'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_55'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeProteinSequence(File file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_56'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;ProteinSequence&#62; proteinSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_57'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_58'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_59'/>
59
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_59'>1</a>
<span>
1. writeProteinSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeProteinSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeProteinSequence(bo, proteinSequences);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_60'/>
60
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_60'>1</a>
<span>
1. writeProteinSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_61'/>
61
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_61'>1</a>
<span>
1. writeProteinSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_62'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_63'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_64'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_65'/>
65
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_65'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write collection of protein sequences to a stream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_66'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_67'/>
67
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_67'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param proteinSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_68'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_69'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_70'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_71'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeProteinSequence(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_72'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;ProteinSequence&#62; proteinSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_73'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_74'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FastaWriter&#60;ProteinSequence, AminoAcidCompound&#62; fastaWriter = new FastaWriter&#60;ProteinSequence, AminoAcidCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_75'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream, proteinSequences,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_76'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericFastaHeaderFormat&#60;ProteinSequence, AminoAcidCompound&#62;());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_77'/>
77
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_77'>1</a>
<span>
1. writeProteinSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;fastaWriter.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_78'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_79'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_80'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_81'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_82'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_83'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_84'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param geneSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_85'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_86'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_87'/>
87
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_87'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_88'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeGeneSequence(File file, Collection&#60;GeneSequence&#62; geneSequences,boolean showExonUppercase) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_89'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_90'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_91'/>
91
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_91'>1</a>
<span>
1. writeGeneSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeGeneSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeGeneSequence(bo, geneSequences,showExonUppercase);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_92'/>
92
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_92'>1</a>
<span>
1. writeGeneSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_93'/>
93
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_93'>1</a>
<span>
1. writeGeneSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_94'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_95'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_96'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_97'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_98'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_99'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_100'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_101'/>
101
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_101'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_102'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_103'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeGeneSequence(OutputStream outputStream, Collection&#60;GeneSequence&#62; geneSequences,boolean showExonUppercase) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_104'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FastaGeneWriter fastaWriter = new FastaGeneWriter(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_105'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream, geneSequences,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_106'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericFastaHeaderFormat&#60;GeneSequence, NucleotideCompound&#62;(),showExonUppercase);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_107'/>
107
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_107'>1</a>
<span>
1. writeGeneSequence : removed call to org/biojava/nbio/core/sequence/io/FastaGeneWriter::process &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;fastaWriter.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_108'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_109'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_110'/>
110
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_110'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_111'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_112'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_113'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of NucleotideSequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_114'/>
114
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_114'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_115'/>
115
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_115'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_116'/>
116
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_116'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_117'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_118'/>
118
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_118'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_119'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeNucleotideSequence(File file, Collection&#60;DNASequence&#62; dnaSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_120'/>
120
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_120'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_121'/>
121
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_121'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_122'/>
122
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_122'>1</a>
<span>
1. writeNucleotideSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeNucleotideSequence(bo, dnaSequences);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_123'/>
123
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_123'>1</a>
<span>
1. writeNucleotideSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_124'/>
124
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_124'>1</a>
<span>
1. writeNucleotideSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_125'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_126'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_127'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_128'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of NucleotideSequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_129'/>
129
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_129'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_130'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_131'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_132'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_133'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_134'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeNucleotideSequence(OutputStream outputStream, Collection&#60;DNASequence&#62; dnaSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_135'/>
135
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_135'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FastaWriter&#60;DNASequence, NucleotideCompound&#62; fastaWriter = new FastaWriter&#60;DNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_136'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream, dnaSequences,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_137'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericFastaHeaderFormat&#60;DNASequence, NucleotideCompound&#62;());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_138'/>
138
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_138'>1</a>
<span>
1. writeNucleotideSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;fastaWriter.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_139'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_140'/>
140
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_140'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_141'/>
141
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_141'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_142'/>
142
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_142'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_143'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a sequence to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_144'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_145'/>
145
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_145'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_146'/>
146
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_146'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_147'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_148'/>
148
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_148'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequence(File file, Sequence&#60;?&#62; sequence) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_149'/>
149
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_149'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_150'/>
150
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_150'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_151'/>
151
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_151'>1</a>
<span>
1. writeSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeSequences &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeSequences(bo, singleSeqToCollection(sequence));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_152'/>
152
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_152'>1</a>
<span>
1. writeSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_153'/>
153
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_153'>1</a>
<span>
1. writeSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_154'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_155'/>
155
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_155'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_156'/>
156
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_156'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_157'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a sequence to OutputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_158'/>
158
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_158'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_159'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_160'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_161'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_162'/>
162
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_162'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequence(OutputStream outputStream, Sequence&#60;?&#62; sequence) throws Exception {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_163'/>
163
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_163'>1</a>
<span>
1. writeSequence : removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeSequences &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;writeSequences(outputStream, singleSeqToCollection(sequence));</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_164'/>
164
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_164'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_165'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_166'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_167'/>
167
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_167'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_168'/>
168
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_168'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_169'/>
169
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_169'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_170'/>
170
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_170'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_171'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_172'/>
172
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_172'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private static Collection&#60;Sequence&#60;?&#62;&#62; singleSeqToCollection(Sequence&#60;?&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_173'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;Collection&#60;Sequence&#60;?&#62;&#62; sequences = new ArrayList&#60;Sequence&#60;?&#62;&#62;();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_174'/>
174
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_174'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;sequences.add(sequence);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_175'/>
175
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_175'>1</a>
<span>
1. singleSeqToCollection : replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/io/FastaWriterHelper::singleSeqToCollection &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return sequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_176'/>
176
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_176'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_177'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_178'/>
178
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_178'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_179'/>
179
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_179'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Method which will write your given Sequences to the specified</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_180'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link OutputStream}. This is a very generic method which writes just the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_181'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * AccessionID of the Sequence as the FASTA header.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_182'/>
182
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_182'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_183'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream Stream to write to; can be System.out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_184'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequences The sequences to write out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_185'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception Thrown normally thanks to IO problems</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_186'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_187'/>
187
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_187'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequences(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_188'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;Sequence&#60;?&#62;&#62; sequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_189'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_190'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;FastaHeaderFormatInterface&#60;Sequence&#60;?&#62;, Compound&#62; fhfi =</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_191'/>
191
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_191'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;new FastaHeaderFormatInterface&#60;Sequence&#60;?&#62;, Compound&#62;() {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_192'/>
192
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_192'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_193'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_194'/>
194
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_194'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;public String getHeader(Sequence&#60;?&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_195'/>
195
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_195'>1</a>
<span>
1. getHeader : replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/io/FastaWriterHelper$1::getHeader &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;&#9;&#9;return sequence.getAccession().toString();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_196'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_197'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_198'/>
198
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_198'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_199'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;};</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_200'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_201'/>
201
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_201'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;FastaWriter&#60;Sequence&#60;?&#62;, Compound&#62; fastaWriter =</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_202'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new FastaWriter&#60;Sequence&#60;?&#62;, Compound&#62;(outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_203'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;sequences, fhfi);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_204'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_205'/>
205
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_205'>1</a>
<span>
1. writeSequences : removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;fastaWriter.process();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_206'/>
206
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_206'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6bc96f76_207'/>
207
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_207'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_59'>59</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_59'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeProteinSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_60'>60</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_60'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_61'>61</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_61'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_77'>77</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_77'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_91'>91</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_91'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeGeneSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeGeneSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_92'>92</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_92'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeGeneSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_93'>93</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_93'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeGeneSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_107'>107</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_107'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeGeneSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaGeneWriter::process &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_122'>122</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_122'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_123'>123</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_123'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_124'>124</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_124'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_138'>138</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_138'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_151'>151</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_151'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeSequences &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_152'>152</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_152'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_153'>153</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_153'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_163'>163</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_163'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriterHelper::writeSequences &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_175'>175</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_175'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>singleSeqToCollection<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/io/FastaWriterHelper::singleSeqToCollection &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_195'>195</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_195'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getHeader<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/io/FastaWriterHelper$1::getHeader &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6bc96f76_205'>205</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6bc96f76_205'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequences<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> removed call to org/biojava/nbio/core/sequence/io/FastaWriter::process &rarr; KILLED</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>BOOLEAN_FALSE_RETURN</li>
<li class='mutator'>BOOLEAN_TRUE_RETURN</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>
<li class='mutator'>EMPTY_RETURN_VALUES</li>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>NULL_RETURN_VALUES</li>
<li class='mutator'>PRIMITIVE_RETURN_VALS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
<li>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeFastaEqualToLineLength(org.biojava.nbio.core.sequence.io.FastaWriterTest) (1 ms)</li>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.5.1

</body>
</html>