<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>GenbankReaderHelper.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_1'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_2'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_3'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_4'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_5'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_6'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_7'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_8'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_9'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_10'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_11'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_12'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_13'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_14'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_15'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_16'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_17'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_18'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_19'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_20'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Created on 01-21-2010</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_21'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_22'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence.io;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_23'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_24'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.DNASequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_25'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.ProteinSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_26'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.RNASequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_27'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_28'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_29'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_30'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_31'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.RNACompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_32'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.AbstractSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_33'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.Logger;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_34'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.LoggerFactory;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_35'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_36'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.File;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_37'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.FileInputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_38'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.InputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_39'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.LinkedHashMap;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_40'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_41'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_42'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_43'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Scooter Willis &#60;willishf at gmail dot com&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_44'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_45'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>public class GenbankReaderHelper {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_46'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_47'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;private final static Logger logger = LoggerFactory.getLogger(GenbankReaderHelper.class);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_48'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_49'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_50'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_51'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_52'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * in one sequence based on accession id.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_53'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_54'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param lazySequenceLoad</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_55'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_56'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_57'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_58'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, DNASequence&#62; readGenbankDNASequence(File file, boolean lazySequenceLoad) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_59'/>
59
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_59'>1</a>
<span>
1. readGenbankDNASequence : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;if (!lazySequenceLoad) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_60'/>
60
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_60'>1</a>
<span>
1. readGenbankDNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;return readGenbankDNASequence(file);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_61'/>
61
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_61'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_62'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_63'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;DNASequence, NucleotideCompound&#62; GenbankProxyReader =</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_64'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenbankReader&#60;DNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_65'/>
65
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_65'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_66'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;DNASequence, NucleotideCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_67'/>
67
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_67'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new FileProxyDNASequenceCreator(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_68'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_69'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;DNACompoundSet.getDNACompoundSet(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_70'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;new GenbankSequenceParser&#60;AbstractSequence&#60;NucleotideCompound&#62;, NucleotideCompound&#62;()</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_71'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_72'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_73'/>
73
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_73'>1</a>
<span>
1. readGenbankDNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return GenbankProxyReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_74'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_75'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_76'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_77'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_78'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_79'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_80'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * in one sequence based on accession id.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_81'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_82'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param lazySequenceLoad</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_83'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_84'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_85'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_86'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, ProteinSequence&#62; readGenbankProteinSequence(File file, boolean lazySequenceLoad) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_87'/>
87
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_87'>1</a>
<span>
1. readGenbankProteinSequence : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;if (!lazySequenceLoad) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_88'/>
88
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_88'>1</a>
<span>
1. readGenbankProteinSequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;return readGenbankProteinSequence(file);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_89'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_90'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_91'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;ProteinSequence, AminoAcidCompound&#62; GenbankProxyReader =</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_92'/>
92
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_92'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenbankReader&#60;ProteinSequence, AminoAcidCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_93'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_94'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;ProteinSequence, AminoAcidCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_95'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new FileProxyProteinSequenceCreator(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_96'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_97'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;AminoAcidCompoundSet.getAminoAcidCompoundSet(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_98'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;new GenbankSequenceParser&#60;AbstractSequence&#60;AminoAcidCompound&#62;, AminoAcidCompound&#62;()</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_99'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_100'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_101'/>
101
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_101'>1</a>
<span>
1. readGenbankProteinSequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return GenbankProxyReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_102'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_103'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_104'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_105'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_106'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_107'/>
107
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_107'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_108'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * in one sequence based on accession id.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_109'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_110'/>
110
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_110'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param lazySequenceLoad</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_111'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_112'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_113'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_114'/>
114
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_114'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, RNASequence&#62; readGenbankRNASequence(File file, boolean lazySequenceLoad) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_115'/>
115
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_115'>1</a>
<span>
1. readGenbankRNASequence : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;if (!lazySequenceLoad) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_116'/>
116
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_116'>1</a>
<span>
1. readGenbankRNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;return readGenbankRNASequence(file);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_117'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_118'/>
118
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_118'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_119'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;RNASequence, NucleotideCompound&#62; GenbankProxyReader =</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_120'/>
120
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_120'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenbankReader&#60;RNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_121'/>
121
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_121'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_122'/>
122
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_122'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;RNASequence, NucleotideCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_123'/>
123
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_123'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;new FileProxyRNASequenceCreator(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_124'/>
124
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_124'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;file,</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_125'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;RNACompoundSet.getRNACompoundSet(),</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_126'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;&#9;new GenbankSequenceParser&#60;AbstractSequence&#60;NucleotideCompound&#62;, NucleotideCompound&#62;()</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_127'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;&#9;)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_128'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_129'/>
129
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_129'>1</a>
<span>
1. readGenbankRNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return GenbankProxyReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_130'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_131'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_132'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_133'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_134'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Read a Genbank file containing amino acids with setup that would handle most</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_135'/>
135
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_135'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * cases.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_136'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_137'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_138'/>
138
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_138'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_139'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_140'/>
140
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_140'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_141'/>
141
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_141'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, ProteinSequence&#62; readGenbankProteinSequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_142'/>
142
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_142'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;File file) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_143'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileInputStream inStream = new FileInputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_144'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String, ProteinSequence&#62; proteinSequences = readGenbankProteinSequence(inStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_145'/>
145
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_145'>1</a>
<span>
1. readGenbankProteinSequence : removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;inStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_146'/>
146
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_146'>1</a>
<span>
1. readGenbankProteinSequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return proteinSequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_147'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_148'/>
148
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_148'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_149'/>
149
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_149'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_150'/>
150
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_150'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Read a Genbank file containing amino acids with setup that would handle most</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_151'/>
151
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_151'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * cases. User is responsible for closing InputStream because you opened it</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_152'/>
152
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_152'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_153'/>
153
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_153'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param inStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_154'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_155'/>
155
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_155'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_156'/>
156
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_156'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_157'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, ProteinSequence&#62; readGenbankProteinSequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_158'/>
158
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_158'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;InputStream inStream) throws Exception {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_159'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;ProteinSequence, AminoAcidCompound&#62; GenbankReader = new GenbankReader&#60;ProteinSequence, AminoAcidCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_160'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;inStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_161'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;ProteinSequence, AminoAcidCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_162'/>
162
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_162'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_163'/>
163
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_163'>1</a>
<span>
1. readGenbankProteinSequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return GenbankReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_164'/>
164
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_164'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_165'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_166'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_167'/>
167
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_167'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Read a Genbank DNA sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_168'/>
168
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_168'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param inStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_169'/>
169
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_169'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_170'/>
170
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_170'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_171'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_172'/>
172
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_172'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, DNASequence&#62; readGenbankDNASequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_173'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;InputStream inStream) throws Exception {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_174'/>
174
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_174'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;DNASequence, NucleotideCompound&#62; GenbankReader = new GenbankReader&#60;DNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_175'/>
175
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_175'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;inStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_176'/>
176
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_176'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;DNASequence, NucleotideCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_177'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_178'/>
178
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_178'>1</a>
<span>
1. readGenbankDNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return GenbankReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_179'/>
179
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_179'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_180'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_181'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_182'/>
182
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_182'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_183'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_184'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_185'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_186'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_187'/>
187
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_187'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, DNASequence&#62; readGenbankDNASequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_188'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;File file) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_189'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileInputStream inStream = new FileInputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_190'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String, DNASequence&#62; dnaSequences = readGenbankDNASequence(inStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_191'/>
191
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_191'>1</a>
<span>
1. readGenbankDNASequence : removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;inStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_192'/>
192
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_192'>1</a>
<span>
1. readGenbankDNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return dnaSequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_193'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_194'/>
194
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_194'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_195'/>
195
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_195'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Read a Genbank RNA sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_196'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param inStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_197'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_198'/>
198
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_198'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_199'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_200'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, RNASequence&#62; readGenbankRNASequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_201'/>
201
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_201'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;InputStream inStream) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_202'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;RNASequence, NucleotideCompound&#62; GenbankReader = new GenbankReader&#60;RNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_203'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;inStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_204'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericGenbankHeaderParser&#60;RNASequence, NucleotideCompound&#62;(),</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_205'/>
205
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_205'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;new RNASequenceCreator(RNACompoundSet.getRNACompoundSet()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_206'/>
206
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_206'>1</a>
<span>
1. readGenbankRNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return GenbankReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_207'/>
207
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_207'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_208'/>
208
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_208'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_209'/>
209
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_209'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_210'/>
210
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_210'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_211'/>
211
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_211'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_212'/>
212
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_212'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_213'/>
213
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_213'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_214'/>
214
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_214'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_215'/>
215
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_215'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static LinkedHashMap&#60;String, RNASequence&#62; readGenbankRNASequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_216'/>
216
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_216'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;File file) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_217'/>
217
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_217'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileInputStream inStream = new FileInputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_218'/>
218
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_218'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String, RNASequence&#62; rnaSequences = readGenbankRNASequence(inStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_219'/>
219
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_219'>1</a>
<span>
1. readGenbankRNASequence : removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;inStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_220'/>
220
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_220'>1</a>
<span>
1. readGenbankRNASequence : replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return rnaSequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_221'/>
221
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_221'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_222'/>
222
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_222'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_223'/>
223
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_223'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void main(String[] args) throws Exception {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_224'/>
224
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_224'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_225'/>
225
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_225'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String, DNASequence&#62; dnaSequences = GenbankReaderHelper.readGenbankDNASequence(new File(&#34;src/test/resources/NM_000266.gb&#34;), true);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_226'/>
226
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_226'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (DNASequence sequence : dnaSequences.values()) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_227'/>
227
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_227'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;logger.info(&#34;DNA Sequence: {}&#34;, sequence.getRNASequence().getProteinSequence().getSequenceAsString());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_228'/>
228
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_228'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_229'/>
229
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_229'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_230'/>
230
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_230'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String, ProteinSequence&#62; proteinSequences = GenbankReaderHelper.readGenbankProteinSequence(new File(&#34;src/test/resources/BondFeature.gb&#34;), true);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_231'/>
231
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_231'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (ProteinSequence sequence : proteinSequences.values()) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_232'/>
232
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_232'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;logger.info(&#34;Protein Sequence: {}&#34;, sequence.getSequenceAsString());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_233'/>
233
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_233'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_234'/>
234
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_234'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@52b074fe_235'/>
235
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_235'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_59'>59</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_59'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_60'>60</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_60'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_73'>73</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_73'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_87'>87</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_87'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_88'>88</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_88'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_101'>101</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_101'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_115'>115</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_115'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_116'>116</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_116'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_129'>129</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_129'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_145'>145</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_145'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_146'>146</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_146'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_163'>163</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_163'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankProteinSequence<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankCookbookTest</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankProteinSequence &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_178'>178</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_178'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankWriterTest.testProcess(org.biojava.nbio.core.sequence.io.GenbankWriterTest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_191'>191</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_191'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_192'>192</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_192'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankDNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankDNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_206'>206</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_206'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_219'>219</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_219'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@52b074fe_220'>220</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@52b074fe_220'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>readGenbankRNASequence<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/io/GenbankReaderHelper::readGenbankRNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>BOOLEAN_FALSE_RETURN</li>
<li class='mutator'>BOOLEAN_TRUE_RETURN</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>
<li class='mutator'>EMPTY_RETURN_VALUES</li>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>NULL_RETURN_VALUES</li>
<li class='mutator'>PRIMITIVE_RETURN_VALS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
<li>org.biojava.nbio.core.sequence.io.GenbankCookbookTest (1675 ms)</li><li>org.biojava.nbio.core.sequence.io.GenbankWriterTest.testProcess(org.biojava.nbio.core.sequence.io.GenbankWriterTest) (10 ms)</li>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.5.1

</body>
</html>