<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>GenbankWriterHelper.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_1'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_2'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_3'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_4'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_5'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_6'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_7'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_8'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_9'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_10'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_11'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_12'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_13'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_14'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_15'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_16'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_17'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_18'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_19'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_20'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_21'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_22'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_23'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_24'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence.io;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_25'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_26'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.DNASequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_27'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.ProteinSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_28'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_29'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_30'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.io.template.GenbankHeaderFormatInterface;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_31'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.Compound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_32'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.Sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_33'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_34'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.BufferedOutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_35'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.File;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_36'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.FileOutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_37'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.OutputStream;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_38'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.ArrayList;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_39'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.Collection;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_40'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_41'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_42'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * The class that should be used to write out genbank file of a sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_43'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * collection</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_44'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_45'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * @author mckeee1</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_46'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_47'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_48'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>public class GenbankWriterHelper {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_49'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static final String LINEAR_DNA = &#34;linear&#34;;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_50'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static final String CIRCULAR_DNA = &#34;circular&#34;;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_51'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_52'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_53'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write collection of protein sequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_54'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_55'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_56'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param proteinSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_57'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_58'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_59'/>
59
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_59'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeProteinSequence(File file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_60'/>
60
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_60'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;ProteinSequence&#62; proteinSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_61'/>
61
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_61'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_62'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_63'/>
63
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_63'>1</a>
<span>
1. writeProteinSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeProteinSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeProteinSequence(bo, proteinSequences);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_64'/>
64
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_64'>1</a>
<span>
1. writeProteinSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_65'/>
65
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_65'>1</a>
<span>
1. writeProteinSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_66'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_67'/>
67
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_67'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_68'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_69'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write collection of protein sequences to a stream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_70'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_71'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_72'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param proteinSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_73'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_74'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_75'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_76'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeProteinSequence(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_77'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;ProteinSequence&#62; proteinSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_78'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_79'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankWriter&#60;ProteinSequence, AminoAcidCompound&#62; genbankWriter = new GenbankWriter&#60;ProteinSequence, AminoAcidCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_80'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_81'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;proteinSequences,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_82'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;new GenericGenbankHeaderFormat&#60;ProteinSequence, AminoAcidCompound&#62;());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_83'/>
83
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_83'>1</a>
<span>
1. writeProteinSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;genbankWriter.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_84'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_85'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_86'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_87'/>
87
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_87'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_88'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of NucleotideSequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_89'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_90'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_91'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_92'/>
92
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_92'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_93'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_94'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_95'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeNucleotideSequence(File file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_96'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;DNASequence&#62; dnaSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_97'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_98'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_99'/>
99
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_99'>1</a>
<span>
1. writeNucleotideSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeNucleotideSequence(bo, dnaSequences);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_100'/>
100
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_100'>1</a>
<span>
1. writeNucleotideSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_101'/>
101
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_101'>1</a>
<span>
1. writeNucleotideSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_102'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_103'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_104'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_105'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of NucleotideSequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_106'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_107'/>
107
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_107'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_108'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_109'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_110'/>
110
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_110'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_111'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_112'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeNucleotideSequence(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_113'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;DNASequence&#62; dnaSequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_114'/>
114
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_114'>1</a>
<span>
1. writeNucleotideSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeNucleotideSequence(outputStream, dnaSequences, LINEAR_DNA);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_115'/>
115
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_115'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_116'/>
116
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_116'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_117'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_118'/>
118
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_118'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a collection of NucleotideSequences to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_119'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_120'/>
120
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_120'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_121'/>
121
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_121'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param dnaSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_122'/>
122
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_122'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param seqType</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_123'/>
123
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_123'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_124'/>
124
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_124'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_125'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_126'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeNucleotideSequence(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_127'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;DNASequence&#62; dnaSequences, String seqType)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_128'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;throws Exception {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_129'/>
129
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_129'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;GenericGenbankHeaderFormat&#60;DNASequence, NucleotideCompound&#62; genericGenbankHeaderFormat = new GenericGenbankHeaderFormat&#60;DNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_130'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;seqType);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_131'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;// genericGenbankHeaderFormat.setLineSeparator(lineSep);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_132'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;GenbankWriter&#60;DNASequence, NucleotideCompound&#62; genbankWriter = new GenbankWriter&#60;DNASequence, NucleotideCompound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_133'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream, dnaSequences, genericGenbankHeaderFormat);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_134'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;// genbankWriter.setLineSeparator(lineSep);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_135'/>
135
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_135'>1</a>
<span>
1. writeNucleotideSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; SURVIVED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;genbankWriter.process();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_136'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_137'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_138'/>
138
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_138'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_139'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a sequence to a file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_140'/>
140
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_140'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_141'/>
141
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_141'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_142'/>
142
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_142'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_143'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_144'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_145'/>
145
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_145'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequence(File file, Sequence&#60;?&#62; sequence)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_146'/>
146
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_146'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_147'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileOutputStream outputStream = new FileOutputStream(file);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_148'/>
148
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_148'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_149'/>
149
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_149'>1</a>
<span>
1. writeSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeSequences &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeSequences(bo, singleSeqToCollection(sequence));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_150'/>
150
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_150'>1</a>
<span>
1. writeSequence : removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;bo.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_151'/>
151
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_151'>1</a>
<span>
1. writeSequence : removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;outputStream.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_152'/>
152
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_152'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_153'/>
153
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_153'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_154'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_155'/>
155
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_155'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Write a sequence to OutputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_156'/>
156
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_156'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_157'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_158'/>
158
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_158'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_159'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_160'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_161'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequence(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_162'/>
162
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_162'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Sequence&#60;?&#62; sequence) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_163'/>
163
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_163'>1</a>
<span>
1. writeSequence : removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeSequences &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;writeSequences(outputStream, singleSeqToCollection(sequence));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_164'/>
164
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_164'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_165'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_166'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_167'/>
167
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_167'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_168'/>
168
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_168'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_169'/>
169
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_169'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_170'/>
170
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_170'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_171'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_172'/>
172
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_172'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private static Collection&#60;Sequence&#60;?&#62;&#62; singleSeqToCollection(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_173'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Sequence&#60;?&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_174'/>
174
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_174'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;Collection&#60;Sequence&#60;?&#62;&#62; sequences = new ArrayList&#60;Sequence&#60;?&#62;&#62;();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_175'/>
175
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_175'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;sequences.add(sequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_176'/>
176
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_176'>1</a>
<span>
1. singleSeqToCollection : replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/io/GenbankWriterHelper::singleSeqToCollection &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return sequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_177'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_178'/>
178
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_178'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_179'/>
179
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_179'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_180'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Method which will write your given Sequences to the specified</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_181'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link OutputStream}. This is a very generic method which writes just the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_182'/>
182
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_182'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * AccessionID of the Sequence as the FASTA header.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_183'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_184'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param outputStream</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_185'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *            Stream to write to; can be System.out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_186'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_187'/>
187
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_187'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *            The sequences to write out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_188'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws Exception</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_189'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *             Thrown normally thanks to IO problems</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_190'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_191'/>
191
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_191'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void writeSequences(OutputStream outputStream,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_192'/>
192
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_192'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Collection&#60;Sequence&#60;?&#62;&#62; sequences) throws Exception {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_193'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_194'/>
194
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_194'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankHeaderFormatInterface&#60;Sequence&#60;?&#62;, Compound&#62; fhfi = new GenbankHeaderFormatInterface&#60;Sequence&#60;?&#62;, Compound&#62;() {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_195'/>
195
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_195'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_196'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_197'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;public String getHeader(Sequence&#60;?&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_198'/>
198
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_198'>1</a>
<span>
1. getHeader : replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/io/GenbankWriterHelper$1::getHeader &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;return sequence.getAccession().toString();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_199'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_200'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_201'/>
201
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_201'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_202'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;};</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_203'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_204'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankWriter&#60;Sequence&#60;?&#62;, Compound&#62; genbankWriter = new GenbankWriter&#60;Sequence&#60;?&#62;, Compound&#62;(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_205'/>
205
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_205'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;outputStream, sequences, fhfi);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_206'/>
206
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_206'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_207'/>
207
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_207'>1</a>
<span>
1. writeSequences : removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;genbankWriter.process();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_208'/>
208
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_208'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@44c7bd26_209'/>
209
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_209'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_63'>63</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_63'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeProteinSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_64'>64</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_64'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_65'>65</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_65'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_83'>83</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_83'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_99'>99</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_99'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_100'>100</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_100'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_101'>101</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_101'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_114'>114</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_114'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeNucleotideSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_135'>135</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_135'/> 

<p class='SURVIVED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeNucleotideSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; SURVIVED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_149'>149</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_149'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeSequences &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_150'>150</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_150'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_151'>151</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_151'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileOutputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_163'>163</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_163'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriterHelper::writeSequences &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_176'>176</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_176'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>singleSeqToCollection<br/><b>Killed by : </b>none</span></span> replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/io/GenbankWriterHelper::singleSeqToCollection &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_198'>198</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_198'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getHeader<br/><b>Killed by : </b>none</span></span> replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/io/GenbankWriterHelper$1::getHeader &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@44c7bd26_207'>207</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@44c7bd26_207'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>writeSequences<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankWriter::process &rarr; NO_COVERAGE</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>BOOLEAN_FALSE_RETURN</li>
<li class='mutator'>BOOLEAN_TRUE_RETURN</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>
<li class='mutator'>EMPTY_RETURN_VALUES</li>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>NULL_RETURN_VALUES</li>
<li class='mutator'>PRIMITIVE_RETURN_VALS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
<li>org.biojava.nbio.core.sequence.io.GenbankWriterTest.testProcess(org.biojava.nbio.core.sequence.io.GenbankWriterTest) (12 ms)</li>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.5.1

</body>
</html>