<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>SequenceMixin.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_1'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_2'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_3'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_4'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_5'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_6'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_7'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_8'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_9'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_10'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_11'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_12'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_13'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_14'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_15'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_16'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_17'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_18'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_19'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_20'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_21'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence.template;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_22'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_23'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_24'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.storage.ArrayListSequenceReader;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_25'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.views.ComplementSequenceView;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_26'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.views.ReversedSequenceView;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_27'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.views.WindowedSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_28'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.util.CRC64Checksum;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_29'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_30'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.IOException;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_31'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.*;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_32'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_33'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_34'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * Provides a set of static methods to be used as static imports when needed</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_35'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * across multiple Sequence implementations but inheritance gets in the way.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_36'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_37'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * It also provides a place to put utility methods whose application can</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_38'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be to a single class of Sequence e.g. {@link NucleotideCompound}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_39'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * {@link Sequence}; or to any Sequence e.g. looking for the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_40'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * {@link #getComposition(Sequence)} or {@link #getDistribution(Sequence)}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_41'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * for any type of Sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_42'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_43'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * All of these methods assume that you can use the {@link Iterable} interface</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_44'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * offered by the implementations of {@link Sequence} to provide all the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_45'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * compounds that implementation allows you to see. Since sequence should know</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_46'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * nothing about its backing stores (apart from calling out to it) this should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_47'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * be true.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_48'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_49'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> * @author ayates</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_50'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_51'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>public class SequenceMixin {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_52'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_53'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_54'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * For the given vargs of compounds this method counts the number of</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_55'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * times those compounds appear in the given sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_56'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_57'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The {@link Sequence} to perform the count on</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_58'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param compounds The compounds to look for</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_59'/>
59
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_59'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound we are looking for</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_60'/>
60
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_60'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The number of times the given compounds appear in this Sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_61'/>
61
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_61'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_62'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; int countCompounds(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_63'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Sequence&#60;C&#62; sequence, C... compounds) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_64'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;int count = 0;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_65'/>
65
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_65'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;Map&#60;C, Integer&#62; compositon = getComposition(sequence);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_66'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for (C compound : compounds) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_67'/>
67
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_67'>1</a>
<span>
1. countCompounds : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if(compositon.containsKey(compound)) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_68'/>
68
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_68'>1</a>
<span>
1. countCompounds : Replaced integer addition with subtraction &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;count = compositon.get(compound) + count;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_69'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_70'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_71'/>
71
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_71'>1</a>
<span>
1. countCompounds : replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countCompounds &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return count;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_72'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_73'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_74'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_75'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Returns the count of GC in the given sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_76'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_77'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The {@link NucleotideCompound} {@link Sequence} to perform</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_78'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the GC analysis on</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_79'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The number of GC compounds in the sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_80'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_81'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static int countGC(Sequence&#60;NucleotideCompound&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_82'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;CompoundSet&#60;NucleotideCompound&#62; cs = sequence.getCompoundSet();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_83'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound G = cs.getCompoundForString(&#34;G&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_84'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound C = cs.getCompoundForString(&#34;C&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_85'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound g = cs.getCompoundForString(&#34;g&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_86'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound c = cs.getCompoundForString(&#34;c&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_87'/>
87
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_87'>1</a>
<span>
1. countGC : replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countGC &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return countCompounds(sequence, G, C, g, c);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_88'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_89'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_90'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_91'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Returns the count of AT in the given sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_92'/>
92
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_92'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_93'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The {@link NucleotideCompound} {@link Sequence} to perform</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_94'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the AT analysis on</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_95'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The number of AT compounds in the sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_96'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_97'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static int countAT(Sequence&#60;NucleotideCompound&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_98'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;CompoundSet&#60;NucleotideCompound&#62; cs = sequence.getCompoundSet();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_99'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound A = cs.getCompoundForString(&#34;A&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_100'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound T = cs.getCompoundForString(&#34;T&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_101'/>
101
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_101'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound a = cs.getCompoundForString(&#34;a&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_102'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;NucleotideCompound t = cs.getCompoundForString(&#34;t&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_103'/>
103
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_103'>1</a>
<span>
1. countAT : replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countAT &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return countCompounds(sequence, A, T, a, t);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_104'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_105'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_106'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_107'/>
107
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_107'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Analogous to {@link #getComposition(Sequence)} but returns the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_108'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * distribution of that {@link Compound} over the given sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_109'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_110'/>
110
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_110'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound to look for</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_111'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The type of sequence to look over</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_112'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return Returns the decimal fraction of the compounds in the given</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_113'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * sequence. Any compound not in the Map will return a fraction of 0.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_114'/>
114
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_114'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_115'/>
115
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_115'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; Map&#60;C, Double&#62; getDistribution(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_116'/>
116
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_116'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;Map&#60;C, Double&#62; results = new HashMap&#60;C, Double&#62;();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_117'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;Map&#60;C, Integer&#62; composition = getComposition(sequence);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_118'/>
118
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_118'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;double length = sequence.getLength();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_119'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for (Map.Entry&#60;C, Integer&#62; entry : composition.entrySet()) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_120'/>
120
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_120'>1</a>
<span>
1. getDistribution : Replaced double division with multiplication &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;double dist = entry.getValue().doubleValue() / length;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_121'/>
121
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_121'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;results.put(entry.getKey(), dist);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_122'/>
122
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_122'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_123'/>
123
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_123'>1</a>
<span>
1. getDistribution : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::getDistribution &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return results;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_124'/>
124
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_124'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_125'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_126'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_127'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Does a linear scan over the given Sequence and records the number of</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_128'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * times each base appears. The returned map will return 0 if a compound</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_129'/>
129
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_129'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * is asked for and the Map has no record of it.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_130'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_131'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound to look for</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_132'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The type of sequence to look over</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_133'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return Counts for the instances of all compounds in the sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_134'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_135'/>
135
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_135'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; Map&#60;C, Integer&#62; getComposition(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_136'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;Map&#60;C, Integer&#62; results = new HashMap&#60;C, Integer&#62;();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_137'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_138'/>
138
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_138'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for (C currentCompound : sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_139'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;Integer currentInteger = results.get(currentCompound);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_140'/>
140
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_140'>1</a>
<span>
1. getComposition : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if ( currentInteger == null)</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_141'/>
141
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_141'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;currentInteger = 0;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_142'/>
142
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_142'>1</a>
<span>
1. getComposition : Replaced integer addition with subtraction &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;currentInteger++;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_143'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;results.put(currentCompound, currentInteger);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_144'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_145'/>
145
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_145'>1</a>
<span>
1. getComposition : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::getComposition &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return results;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_146'/>
146
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_146'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_147'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_148'/>
148
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_148'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_149'/>
149
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_149'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Used as a way of sending a Sequence to a writer without the cost of</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_150'/>
150
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_150'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * converting to a full length String and then writing the data out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_151'/>
151
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_151'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_152'/>
152
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_152'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; Type of compound</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_153'/>
153
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_153'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param writer The writer to send data to</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_154'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The sequence to write out</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_155'/>
155
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_155'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws IOException Thrown if we encounter a problem</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_156'/>
156
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_156'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_157'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; void write(Appendable appendable, Sequence&#60;C&#62; sequence) throws IOException {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_158'/>
158
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_158'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for(C compound: sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_159'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;appendable.append(compound.toString());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_160'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_161'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_162'/>
162
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_162'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_163'/>
163
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_163'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_164'/>
164
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_164'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * For the given Sequence this will return a {@link StringBuilder} object</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_165'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * filled with the results of {@link Compound#toString()}. Does not</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_166'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * used {@link #write(java.lang.Appendable, org.biojava.nbio.core.sequence.template.Sequence) }</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_167'/>
167
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_167'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * because of its {@link IOException} signature.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_168'/>
168
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_168'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_169'/>
169
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_169'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; StringBuilder toStringBuilder(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_170'/>
170
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_170'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;StringBuilder sb = new StringBuilder(sequence.getLength());</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_171'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for (C compound : sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_172'/>
172
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_172'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;sb.append(compound.toString());</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_173'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_174'/>
174
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_174'>1</a>
<span>
1. toStringBuilder : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::toStringBuilder &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return sb;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_175'/>
175
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_175'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_176'/>
176
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_176'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_177'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_178'/>
178
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_178'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Shortcut to {@link #toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_179'/>
179
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_179'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * which calls toString() on the resulting object.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_180'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_181'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; String toString(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_182'/>
182
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_182'>1</a>
<span>
1. toString : replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/template/SequenceMixin::toString &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return toStringBuilder(sequence).toString();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_183'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_184'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_185'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_186'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * For the given {@link Sequence} this will return a {@link List} filled with</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_187'/>
187
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_187'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the Compounds of that {@link Sequence}.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_188'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_189'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; List&#60;C&#62; toList(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_190'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;List&#60;C&#62; list = new ArrayList&#60;C&#62;(sequence.getLength());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_191'/>
191
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_191'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (C compound : sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_192'/>
192
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_192'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;list.add(compound);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_193'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_194'/>
194
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_194'>1</a>
<span>
1. toList : replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::toList &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return list;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_195'/>
195
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_195'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_196'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_197'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_198'/>
198
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_198'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Performs a linear search of the given Sequence for the given compound.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_199'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Once we find the compound we return the position.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_200'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_201'/>
201
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_201'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; int indexOf(Sequence&#60;C&#62; sequence,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_202'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;C compound) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_203'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;int index = 1;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_204'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for (C currentCompound : sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_205'/>
205
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_205'>1</a>
<span>
1. indexOf : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if (currentCompound.equals(compound)) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_206'/>
206
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_206'>1</a>
<span>
1. indexOf : replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::indexOf &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;return index;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_207'/>
207
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_207'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_208'/>
208
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_208'>1</a>
<span>
1. indexOf : Changed increment from 1 to -1 &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;index++;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_209'/>
209
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_209'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_210'/>
210
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_210'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;return 0;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_211'/>
211
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_211'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_212'/>
212
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_212'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_213'/>
213
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_213'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_214'/>
214
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_214'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Performs a reversed linear search of the given Sequence by wrapping</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_215'/>
215
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_215'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * it in a {@link ReversedSequenceView} and passing it into</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_216'/>
216
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_216'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link #indexOf(Sequence, Compound)}. We then inverse the index coming</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_217'/>
217
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_217'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * out of it.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_218'/>
218
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_218'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_219'/>
219
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_219'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; int lastIndexOf(Sequence&#60;C&#62; sequence,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_220'/>
220
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_220'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;C compound) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_221'/>
221
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_221'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;int index = indexOf(new ReversedSequenceView&#60;C&#62;(sequence), compound);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_222'/>
222
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_222'>3</a>
<span>
1. lastIndexOf : Replaced integer subtraction with addition &rarr; KILLED<br/>
2. lastIndexOf : Replaced integer addition with subtraction &rarr; KILLED<br/>
3. lastIndexOf : replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::lastIndexOf &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return (sequence.getLength() - index)+1;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_223'/>
223
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_223'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_224'/>
224
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_224'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_225'/>
225
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_225'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_226'/>
226
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_226'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Creates a simple sequence iterator which moves through a sequence going</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_227'/>
227
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_227'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * from 1 to the length of the Sequence. Modification of the Sequence is not</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_228'/>
228
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_228'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * allowed.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_229'/>
229
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_229'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_230'/>
230
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_230'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; Iterator&#60;C&#62; createIterator(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_231'/>
231
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_231'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_232'/>
232
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_232'>1</a>
<span>
1. createIterator : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::createIterator &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return new SequenceIterator&#60;C&#62;(sequence);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_233'/>
233
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_233'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_234'/>
234
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_234'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_235'/>
235
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_235'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_236'/>
236
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_236'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Creates a simple sub sequence view delimited by the given start and end.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_237'/>
237
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_237'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_238'/>
238
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_238'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; SequenceView&#60;C&#62; createSubSequence(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_239'/>
239
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_239'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;Sequence&#60;C&#62; sequence, int start, int end) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_240'/>
240
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_240'>1</a>
<span>
1. createSubSequence : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::createSubSequence &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return new SequenceProxyView&#60;C&#62;(sequence, start, end);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_241'/>
241
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_241'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_242'/>
242
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_242'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_243'/>
243
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_243'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_244'/>
244
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_244'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Implements sequence shuffling by first materializing the given</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_245'/>
245
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_245'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link Sequence} into a {@link List}, applying</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_246'/>
246
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_246'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link Collections#shuffle(List)} and then returning the shuffled</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_247'/>
247
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_247'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * elements in a new instance of {@link SequenceBackingStore} which behaves</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_248'/>
248
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_248'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * as a {@link Sequence}.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_249'/>
249
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_249'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_250'/>
250
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_250'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; Sequence&#60;C&#62; shuffle(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_251'/>
251
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_251'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;List&#60;C&#62; compounds = sequence.getAsList();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_252'/>
252
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_252'>1</a>
<span>
1. shuffle : removed call to java/util/Collections::shuffle &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;Collections.shuffle(compounds);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_253'/>
253
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_253'>1</a>
<span>
1. shuffle : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::shuffle &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return new ArrayListSequenceReader&#60;C&#62;(compounds,</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_254'/>
254
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_254'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;sequence.getCompoundSet());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_255'/>
255
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_255'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_256'/>
256
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_256'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_257'/>
257
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_257'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_258'/>
258
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_258'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Performs a simple CRC64 checksum on any given sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_259'/>
259
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_259'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_260'/>
260
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_260'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; String checksum(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_261'/>
261
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_261'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;CRC64Checksum checksum = new CRC64Checksum();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_262'/>
262
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_262'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (C compound : sequence) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_263'/>
263
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_263'>1</a>
<span>
1. checksum : removed call to org/biojava/nbio/core/util/CRC64Checksum::update &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;checksum.update(compound.getShortName());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_264'/>
264
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_264'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_265'/>
265
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_265'>1</a>
<span>
1. checksum : replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/template/SequenceMixin::checksum &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return checksum.toString();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_266'/>
266
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_266'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_267'/>
267
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_267'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_268'/>
268
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_268'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_269'/>
269
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_269'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Produces kmers of the specified size e.g. ATGTGA returns two views which</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_270'/>
270
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_270'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * have ATG TGA</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_271'/>
271
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_271'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_272'/>
272
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_272'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; Compound to use</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_273'/>
273
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_273'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence Sequence to build from</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_274'/>
274
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_274'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param kmer Kmer size</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_275'/>
275
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_275'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The list of non-overlapping K-mers</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_276'/>
276
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_276'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_277'/>
277
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_277'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; List&#60;SequenceView&#60;C&#62;&#62; nonOverlappingKmers(Sequence&#60;C&#62; sequence, int kmer) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_278'/>
278
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_278'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;List&#60;SequenceView&#60;C&#62;&#62; l = new ArrayList&#60;SequenceView&#60;C&#62;&#62;();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_279'/>
279
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_279'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;WindowedSequence&#60;C&#62; w = new WindowedSequence&#60;C&#62;(sequence, kmer);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_280'/>
280
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_280'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;for(SequenceView&#60;C&#62; view: w) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_281'/>
281
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_281'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;l.add(view);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_282'/>
282
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_282'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_283'/>
283
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_283'>1</a>
<span>
1. nonOverlappingKmers : replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::nonOverlappingKmers &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return l;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_284'/>
284
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_284'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_285'/>
285
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_285'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_286'/>
286
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_286'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_287'/>
287
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_287'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Used to generate overlapping k-mers such i.e. ATGTA will give rise to</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_288'/>
288
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_288'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * ATG, TGT &#38; GTA</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_289'/>
289
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_289'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_290'/>
290
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_290'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; Compound to use</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_291'/>
291
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_291'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence Sequence to build from</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_292'/>
292
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_292'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param kmer Kmer size</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_293'/>
293
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_293'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The list of overlapping K-mers</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_294'/>
294
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_294'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_295'/>
295
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_295'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; List&#60;SequenceView&#60;C&#62;&#62; overlappingKmers(Sequence&#60;C&#62; sequence, int kmer) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_296'/>
296
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_296'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;List&#60;SequenceView&#60;C&#62;&#62; l = new ArrayList&#60;SequenceView&#60;C&#62;&#62;();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_297'/>
297
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_297'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;List&#60;Iterator&#60;SequenceView&#60;C&#62;&#62;&#62; windows</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_298'/>
298
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_298'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;= new ArrayList&#60;Iterator&#60;SequenceView&#60;C&#62;&#62;&#62;();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_299'/>
299
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_299'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_300'/>
300
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_300'>3</a>
<span>
1. overlappingKmers : changed conditional boundary &rarr; KILLED<br/>
2. overlappingKmers : Changed increment from 1 to -1 &rarr; KILLED<br/>
3. overlappingKmers : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;for(int i=1; i&#60;=kmer; i++) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_301'/>
301
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_301'>1</a>
<span>
1. overlappingKmers : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if(i == 1) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_302'/>
302
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_302'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;windows.add(new WindowedSequence&#60;C&#62;(sequence, kmer).iterator());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_303'/>
303
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_303'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_304'/>
304
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_304'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;else {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_305'/>
305
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_305'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;SequenceView&#60;C&#62; sv = sequence.getSubSequence(i, sequence.getLength());</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_306'/>
306
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_306'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;windows.add(new WindowedSequence&#60;C&#62;(sv, kmer).iterator());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_307'/>
307
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_307'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_308'/>
308
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_308'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_309'/>
309
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_309'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_310'/>
310
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_310'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;OUTER: while(true) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_311'/>
311
</td>
<td class='uncertain'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_311'>3</a>
<span>
1. overlappingKmers : negated conditional &rarr; TIMED_OUT<br/>
2. overlappingKmers : changed conditional boundary &rarr; KILLED<br/>
3. overlappingKmers : Changed increment from 1 to -1 &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='uncertain'>&#9;&#9;&#9;for(int i=0; i&#60;kmer; i++) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_312'/>
312
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_312'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;Iterator&#60;SequenceView&#60;C&#62;&#62; iterator = windows.get(i);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_313'/>
313
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_313'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;boolean breakLoop=true;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_314'/>
314
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_314'>1</a>
<span>
1. overlappingKmers : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;if(iterator.hasNext()) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_315'/>
315
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_315'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;l.add(iterator.next());</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_316'/>
316
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_316'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;breakLoop = false;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_317'/>
317
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_317'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_318'/>
318
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_318'>1</a>
<span>
1. overlappingKmers : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;if(breakLoop) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_319'/>
319
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_319'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;break OUTER;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_320'/>
320
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_320'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_321'/>
321
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_321'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_322'/>
322
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_322'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_323'/>
323
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_323'>1</a>
<span>
1. overlappingKmers : replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::overlappingKmers &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return l;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_324'/>
324
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_324'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_325'/>
325
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_325'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_326'/>
326
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_326'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_327'/>
327
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_327'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * A method which attempts to do the right thing when is comes to a</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_328'/>
328
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_328'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * reverse/reverse complement</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_329'/>
329
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_329'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_330'/>
330
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_330'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_331'/>
331
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_331'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequence The input sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_332'/>
332
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_332'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return The inverted sequence which is optionally complemented</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_333'/>
333
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_333'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_334'/>
334
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_334'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;@SuppressWarnings({ &#34;unchecked&#34; })</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_335'/>
335
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_335'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; SequenceView&#60;C&#62; inverse(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_336'/>
336
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_336'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;SequenceView&#60;C&#62; reverse = new ReversedSequenceView&#60;C&#62;(sequence);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_337'/>
337
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_337'>1</a>
<span>
1. inverse : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;if(sequence.getCompoundSet().isComplementable()) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_338'/>
338
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_338'>1</a>
<span>
1. inverse : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::inverse &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;return new ComplementSequenceView(reverse);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_339'/>
339
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_339'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_340'/>
340
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_340'>1</a>
<span>
1. inverse : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::inverse &rarr; NO_COVERAGE<br/>

</span>
</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return reverse;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_341'/>
341
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_341'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_342'/>
342
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_342'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_343'/>
343
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_343'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_344'/>
344
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_344'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * A case-insensitive manner of comparing two sequence objects together.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_345'/>
345
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_345'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * We will throw out any compounds which fail to match on their sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_346'/>
346
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_346'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * length &#38; compound sets used. The code will also bail out the moment</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_347'/>
347
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_347'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * we find something is wrong with a Sequence. Cost to run is linear to</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_348'/>
348
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_348'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the length of the Sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_349'/>
349
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_349'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_350'/>
350
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_350'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_351'/>
351
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_351'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param source Source sequence to assess</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_352'/>
352
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_352'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param target Target sequence to assess</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_353'/>
353
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_353'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return Boolean indicating if the sequences matched ignoring case</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_354'/>
354
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_354'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_355'/>
355
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_355'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; boolean sequenceEqualityIgnoreCase(Sequence&#60;C&#62; source, Sequence&#60;C&#62; target) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_356'/>
356
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_356'>2</a>
<span>
1. sequenceEqualityIgnoreCase : replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEqualityIgnoreCase &rarr; SURVIVED<br/>
2. sequenceEqualityIgnoreCase : replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEqualityIgnoreCase &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;return baseSequenceEquality(source, target, true);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_357'/>
357
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_357'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_358'/>
358
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_358'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_359'/>
359
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_359'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_360'/>
360
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_360'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * A case-sensitive manner of comparing two sequence objects together.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_361'/>
361
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_361'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * We will throw out any compounds which fail to match on their sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_362'/>
362
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_362'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * length &#38; compound sets used. The code will also bail out the moment</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_363'/>
363
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_363'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * we find something is wrong with a Sequence. Cost to run is linear to</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_364'/>
364
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_364'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the length of the Sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_365'/>
365
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_365'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_366'/>
366
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_366'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; The type of compound</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_367'/>
367
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_367'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param source Source sequence to assess</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_368'/>
368
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_368'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param target Target sequence to assess</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_369'/>
369
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_369'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return Boolean indicating if the sequences matched</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_370'/>
370
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_370'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_371'/>
371
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_371'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static &#60;C extends Compound&#62; boolean sequenceEquality(Sequence&#60;C&#62; source, Sequence&#60;C&#62; target) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_372'/>
372
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_372'>2</a>
<span>
1. sequenceEquality : replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEquality &rarr; KILLED<br/>
2. sequenceEquality : replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEquality &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return baseSequenceEquality(source, target, false);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_373'/>
373
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_373'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_374'/>
374
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_374'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_375'/>
375
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_375'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private static &#60;C extends Compound&#62; boolean baseSequenceEquality(Sequence&#60;C&#62; source, Sequence&#60;C&#62; target, boolean ignoreCase) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_376'/>
376
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_376'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;boolean equal = true;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_377'/>
377
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_377'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;if(</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_378'/>
378
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_378'>1</a>
<span>
1. baseSequenceEquality : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;source.getLength() == target.getLength() &#38;&#38;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_379'/>
379
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_379'>1</a>
<span>
1. baseSequenceEquality : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;source.getCompoundSet().equals(target.getCompoundSet())) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_380'/>
380
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_380'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;Iterator&#60;C&#62; sIter = source.iterator();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_381'/>
381
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_381'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;Iterator&#60;C&#62; tIter = target.iterator();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_382'/>
382
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_382'>1</a>
<span>
1. baseSequenceEquality : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;while(sIter.hasNext()) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_383'/>
383
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_383'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;C s = sIter.next();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_384'/>
384
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_384'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;C t = tIter.next();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_385'/>
385
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_385'>1</a>
<span>
1. baseSequenceEquality : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;boolean cEqual = (ignoreCase) ? s.equalsIgnoreCase(t) : s.equals(t);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_386'/>
386
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_386'>1</a>
<span>
1. baseSequenceEquality : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;if(!cEqual) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_387'/>
387
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_387'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;equal = false;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_388'/>
388
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_388'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;break;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_389'/>
389
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_389'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_390'/>
390
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_390'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_391'/>
391
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_391'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_392'/>
392
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_392'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;else {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_393'/>
393
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_393'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;equal = false;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_394'/>
394
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_394'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_395'/>
395
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_395'>2</a>
<span>
1. baseSequenceEquality : replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::baseSequenceEquality &rarr; KILLED<br/>
2. baseSequenceEquality : replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::baseSequenceEquality &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return equal;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_396'/>
396
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_396'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_397'/>
397
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_397'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_398'/>
398
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_398'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_399'/>
399
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_399'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * A basic sequence iterator which iterates over the given Sequence by</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_400'/>
400
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_400'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * biological index. This assumes your sequence supports random access</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_401'/>
401
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_401'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * and performs well when doing these operations.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_402'/>
402
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_402'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_403'/>
403
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_403'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @author ayates</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_404'/>
404
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_404'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_405'/>
405
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_405'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param &#60;C&#62; Type of compound to return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_406'/>
406
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_406'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_407'/>
407
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_407'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static class SequenceIterator&#60;C extends Compound&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_408'/>
408
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_408'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;implements Iterator&#60;C&#62; {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_409'/>
409
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_409'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_410'/>
410
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_410'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;private final Sequence&#60;C&#62; sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_411'/>
411
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_411'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;private final int length;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_412'/>
412
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_412'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;private int currentPosition = 0;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_413'/>
413
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_413'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_414'/>
414
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_414'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;public SequenceIterator(Sequence&#60;C&#62; sequence) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_415'/>
415
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_415'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;this.sequence = sequence;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_416'/>
416
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_416'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;this.length = sequence.getLength();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_417'/>
417
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_417'></a>
<span>
</span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_418'/>
418
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_418'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_419'/>
419
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_419'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_420'/>
420
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_420'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_421'/>
421
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_421'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;public boolean hasNext() {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_422'/>
422
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_422'>3</a>
<span>
1. hasNext : replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin$SequenceIterator::hasNext &rarr; KILLED<br/>
2. hasNext : changed conditional boundary &rarr; KILLED<br/>
3. hasNext : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;return (currentPosition &#60; length);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_423'/>
423
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_423'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_424'/>
424
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_424'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_425'/>
425
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_425'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_426'/>
426
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_426'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_427'/>
427
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_427'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;public C next() {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_428'/>
428
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_428'>1</a>
<span>
1. next : negated conditional &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if(!hasNext()) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_429'/>
429
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_429'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;throw new NoSuchElementException(&#34;Exhausted sequence of elements&#34;);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_430'/>
430
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_430'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_431'/>
431
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_431'>2</a>
<span>
1. next : Replaced integer addition with subtraction &rarr; KILLED<br/>
2. next : replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin$SequenceIterator::next &rarr; KILLED<br/>

</span>
</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;return sequence.getCompoundAt(++currentPosition);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_432'/>
432
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_432'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_433'/>
433
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_433'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_434'/>
434
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_434'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_435'/>
435
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_435'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;public void remove() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_436'/>
436
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_436'></a>
<span>
</span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;throw new UnsupportedOperationException(&#34;Cannot remove() on a SequenceIterator&#34;);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_437'/>
437
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_437'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_438'/>
438
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_438'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_439'/>
439
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_439'></a>
<span>
</span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_67'>67</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_67'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>countCompounds<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_68'>68</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_68'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>countCompounds<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> Replaced integer addition with subtraction &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_71'>71</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_71'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>countCompounds<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countCompounds &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_87'>87</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_87'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>countGC<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countGC &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_103'>103</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_103'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>countAT<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.at(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::countAT &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_120'>120</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_120'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getDistribution<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.composition(org.biojava.nbio.core.sequence.DNATest)</span></span> Replaced double division with multiplication &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_123'>123</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_123'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getDistribution<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.composition(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::getDistribution &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_140'>140</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_140'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getComposition<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_142'>142</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_142'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getComposition<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> Replaced integer addition with subtraction &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_145'>145</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_145'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getComposition<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::getComposition &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_174'>174</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_174'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>toStringBuilder<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeFastaEqualToLineLength(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::toStringBuilder &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_182'>182</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_182'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>toString<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeFastaEqualToLineLength(org.biojava.nbio.core.sequence.io.FastaWriterTest)</span></span> replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/template/SequenceMixin::toString &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_194'>194</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_194'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>toList<br/><b>Killed by : </b>none</span></span> replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::toList &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_205'>205</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_205'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>indexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_206'>206</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_206'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>indexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::indexOf &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_208'>208</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_208'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>indexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.reverse(org.biojava.nbio.core.sequence.DNATest)</span></span> Changed increment from 1 to -1 &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_222'>222</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_222'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>lastIndexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> Replaced integer subtraction with addition &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>lastIndexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> Replaced integer addition with subtraction &rarr; KILLED</p> <p class='KILLED'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>lastIndexOf<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> replaced int return with 0 for org/biojava/nbio/core/sequence/template/SequenceMixin::lastIndexOf &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_232'>232</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_232'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>createIterator<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.fourBit(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::createIterator &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_240'>240</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_240'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>createSubSequence<br/><b>Killed by : </b>org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest.testGetSequenceAsStringIntegerIntegerStrand(org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::createSubSequence &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_252'>252</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_252'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>shuffle<br/><b>Killed by : </b>none</span></span> removed call to java/util/Collections::shuffle &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_253'>253</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_253'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>shuffle<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::shuffle &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_263'>263</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_263'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>checksum<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/util/CRC64Checksum::update &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_265'>265</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_265'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>checksum<br/><b>Killed by : </b>none</span></span> replaced return value with &#34;&#34; for org/biojava/nbio/core/sequence/template/SequenceMixin::checksum &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_283'>283</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_283'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>nonOverlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerNonOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::nonOverlappingKmers &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_300'>300</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_300'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> changed conditional boundary &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> Changed increment from 1 to -1 &rarr; KILLED</p> <p class='KILLED'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_301'>301</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_301'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_311'>311</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_311'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> changed conditional boundary &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> Changed increment from 1 to -1 &rarr; KILLED</p> <p class='TIMED_OUT'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; TIMED_OUT</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_314'>314</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_314'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_318'>318</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_318'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_323'>323</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_323'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>overlappingKmers<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced return value with Collections.emptyList for org/biojava/nbio/core/sequence/template/SequenceMixin::overlappingKmers &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_337'>337</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_337'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>inverse<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_338'>338</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_338'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>inverse<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::inverse &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_340'>340</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_340'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>inverse<br/><b>Killed by : </b>none</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin::inverse &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_356'>356</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_356'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>sequenceEqualityIgnoreCase<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEqualityIgnoreCase &rarr; KILLED</p> <p class='SURVIVED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>sequenceEqualityIgnoreCase<br/><b>Killed by : </b>none</span></span> replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEqualityIgnoreCase &rarr; SURVIVED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_372'>372</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_372'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>sequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEquality &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>sequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::sequenceEquality &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_378'>378</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_378'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_379'>379</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_379'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_382'>382</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_382'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_385'>385</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_385'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_386'>386</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_386'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_395'>395</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_395'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced boolean return with false for org/biojava/nbio/core/sequence/template/SequenceMixin::baseSequenceEquality &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>baseSequenceEquality<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest)</span></span> replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin::baseSequenceEquality &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_422'>422</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_422'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>hasNext<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> replaced boolean return with true for org/biojava/nbio/core/sequence/template/SequenceMixin$SequenceIterator::hasNext &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>hasNext<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> changed conditional boundary &rarr; KILLED</p> <p class='KILLED'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>hasNext<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_428'>428</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_428'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>next<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@51dcb1c5_431'>431</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@51dcb1c5_431'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>next<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> Replaced integer addition with subtraction &rarr; KILLED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>next<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest)</span></span> replaced return value with null for org/biojava/nbio/core/sequence/template/SequenceMixin$SequenceIterator::next &rarr; KILLED</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>BOOLEAN_FALSE_RETURN</li>
<li class='mutator'>BOOLEAN_TRUE_RETURN</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>
<li class='mutator'>EMPTY_RETURN_VALUES</li>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>NULL_RETURN_VALUES</li>
<li class='mutator'>PRIMITIVE_RETURN_VALS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
<li>org.biojava.nbio.core.sequence.DNATest.kmerOverlapExceedingSequenceLength(org.biojava.nbio.core.sequence.DNATest) (2 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testProteinSequenceFactoring[5](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (85793 ms)</li><li>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeBasicFasta(org.biojava.nbio.core.sequence.io.FastaWriterTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.reverse(org.biojava.nbio.core.sequence.DNATest) (3 ms)</li><li>org.biojava.nbio.core.sequence.location.LocationTest.testBasicCircularLocation(org.biojava.nbio.core.sequence.location.LocationTest) (87 ms)</li><li>org.biojava.nbio.core.sequence.io.FastaWriterTest.writeFastaEqualToLineLength(org.biojava.nbio.core.sequence.io.FastaWriterTest) (0 ms)</li><li>org.biojava.nbio.core.sequence.io.GenbankWriterTest.testProcess(org.biojava.nbio.core.sequence.io.GenbankWriterTest) (12 ms)</li><li>org.biojava.nbio.core.sequence.EditSequenceTest.substitute(org.biojava.nbio.core.sequence.EditSequenceTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.SequenceViewTest.testLastIndexOf(org.biojava.nbio.core.sequence.SequenceViewTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.respectCase(org.biojava.nbio.core.sequence.DNATest) (2 ms)</li><li>org.biojava.nbio.core.sequence.io.ABITracerTest (4399 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testFeatures[1](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (28789 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.gc(org.biojava.nbio.core.sequence.DNATest) (2 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testGetAlignedSequencesSArray(org.biojava.nbio.core.alignment.SimpleProfileTest) (6 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testIterator(org.biojava.nbio.core.alignment.SimpleProfileTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.TranslationTest (2036 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.composition(org.biojava.nbio.core.sequence.DNATest) (3 ms)</li><li>org.biojava.nbio.core.search.io.HspTest (13 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.subSequence(org.biojava.nbio.core.sequence.DNATest) (3 ms)</li><li>org.biojava.nbio.core.sequence.io.GenbankCookbookTest (2366 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.fourBit(org.biojava.nbio.core.sequence.DNATest) (3 ms)</li><li>org.biojava.nbio.core.sequence.SequenceViewTest.testInverse(org.biojava.nbio.core.sequence.SequenceViewTest) (0 ms)</li><li>org.biojava.nbio.core.sequence.location.LocationTest.testSubLocations(org.biojava.nbio.core.sequence.location.LocationTest) (2 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testFeatures[2](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (29139 ms)</li><li>org.biojava.nbio.core.sequence.JoiningSequenceReaderTest.canScan(org.biojava.nbio.core.sequence.JoiningSequenceReaderTest) (0 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testFeatures[5](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (49669 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.kmerOverlap(org.biojava.nbio.core.sequence.DNATest) (2 ms)</li><li>org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest.testGetSequenceAsString(org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest) (9 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.translateToRna(org.biojava.nbio.core.sequence.DNATest) (51 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testToString(org.biojava.nbio.core.alignment.SimpleProfileTest) (6 ms)</li><li>org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest.testToString(org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest) (2 ms)</li><li>org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest.testGetSequenceAsStringIntegerIntegerStrand(org.biojava.nbio.core.alignment.SimpleAlignedSequenceTest) (3 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testGetAlignedSequenceS(org.biojava.nbio.core.alignment.SimpleProfileTest) (2 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testProteinSequenceFactoring[2](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (56652 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testProteinSequenceFactoring[1](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (12091 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.singleCompoundSequence(org.biojava.nbio.core.sequence.DNATest) (5 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testToStringFormatted(org.biojava.nbio.core.alignment.SimpleProfileTest) (18 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.kmerNonOverlap(org.biojava.nbio.core.sequence.DNATest) (5 ms)</li><li>org.biojava.nbio.core.sequence.JoiningSequenceReaderTest.empty(org.biojava.nbio.core.sequence.JoiningSequenceReaderTest) (0 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.twoBit(org.biojava.nbio.core.sequence.DNATest) (23 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testGetAlignedSequenceInt(org.biojava.nbio.core.alignment.SimpleProfileTest) (9 ms)</li><li>org.biojava.nbio.core.sequence.EditSequenceTest.insert(org.biojava.nbio.core.sequence.EditSequenceTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.io.FastaReaderTest (139 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testProteinSequenceFactoring[0](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (79949 ms)</li><li>org.biojava.nbio.core.sequence.EditSequenceTest.delete(org.biojava.nbio.core.sequence.EditSequenceTest) (32 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testToStringInt(org.biojava.nbio.core.alignment.SimpleProfileTest) (3 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.complement(org.biojava.nbio.core.sequence.DNATest) (3 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.reverseComplement(org.biojava.nbio.core.sequence.DNATest) (8 ms)</li><li>org.biojava.nbio.core.sequence.location.LocationTest.testWithStrandSwitch(org.biojava.nbio.core.sequence.location.LocationTest) (1 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.sequenceEquality(org.biojava.nbio.core.sequence.DNATest) (28 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testGetAlignedSequences(org.biojava.nbio.core.alignment.SimpleProfileTest) (2 ms)</li><li>org.biojava.nbio.core.sequence.DNATest.at(org.biojava.nbio.core.sequence.DNATest) (12 ms)</li><li>org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest.testFeatures[0](org.biojava.nbio.core.sequence.loader.GenbankProxySequenceReaderTest) (70063 ms)</li><li>org.biojava.nbio.core.alignment.SimpleProfileTest.testGetAlignedSequencesIntArray(org.biojava.nbio.core.alignment.SimpleProfileTest) (2 ms)</li>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.5.1

</body>
</html>